src/firebolt/common/base_cursor.py

```diff
@@ -273,25 +305,37 @@ def _reset(self) -> None:
         self._next_set_idx = 0
         self._query_id = ""
 
-    def _parse_response_headers(self, headers: Headers) -> None:
-        """Parse response and update relevant cursor fields."""
-        update_parameters = headers.get("Firebolt-Update-Parameters")
-        # parse update parameters dict and set keys as attributes
-        if update_parameters:
-            # parse key1=value1,key2=value2 comma separated string into dict
-            param_dict = dict(item.split("=") for item in update_parameters.split(","))
-            # strip whitespace from keys and values
-            param_dict = {
-                key.strip(): value.strip() for key, value in param_dict.items()
-            }
-            for key, value in param_dict.items():
-                if key in SERVER_SIDE_PARAMETERS:
-                    self.parameters[key] = value
-                else:
-                    logger.debug(
-                        f"Unknown parameter {key} returned by the server. "
-                        "It will be ignored."
-                    )
+    def _update_set_parameters(self, parameters: Dict[str, Any]) -> None:
+        for key, value in parameters.items():
+            if key not in DISALLOWED_PARAMETER_LIST:
+                self._set_parameters[key] = value
+            else:
+                # This should never happen as user parameters are validated
+                # before they are sent to the server
+                logger.debug(
+                    f"Trying to set a disalloed parameter {key} from server. "
+                    "It will be ignored."
+                )
+
+    def _update_server_parameters(self, parameters: Dict[str, Any]) -> None:
+        for key, value in parameters.items():
+            if key in SERVER_SIDE_PARAMETERS:
+                self.parameters[key] = value
+            else:
+                logger.debug(
+                    f"Trying to set an unknown parameter {key}. " "It will be ignored."
+                )
+
+    def get_engine_name(self, engine_url: str) -> str:
+        """
+        Get the name of the engine that we're using.
+
+        Args:
+            engine_url (str): URL of the engine
+        """
+        if self._set_parameters.get("engine"):
```

Comment:
Can we have `engine` in _set_parameters?

Reply:
In case we can't, convert this function into a property

Reply:
Yes, _set_parameters populated from a "Firebolt-Update-Endpoint" header returned when `USE ENGINE` is called, returning the gateway URL and engine=X parameter which specifies which engine we're using.

Reply:
Oh, ok. But in this case am I right that Firebolt-Reset-Session header would also remove the `engine` parameter from _set_statements?

Reply:
That was exactly my q in https://docs.google.com/document/d/1f0kdPCPbhCrmMTK-sfqAkqEkonT7xyNK-teJmXE_6p0/edit?disco=AAABFIz4oXU
Do you think we should make "engine" a special case?

Reply:
Ok, thinking about it it really should be "special". Reworked this part a little.

Reply:
Reworked again, as discussed offline. switched _set_parameters to parameters and added a test for this.